### ISSUE TITLE
json-read: JSON readtable error: 87

### ISSUE THREAD

**Issue by yefeiyu** [millejoh/emacs-ipython-notebook#555]
How should I do?
The OS is `termux`.
python: 3.7.3
emacs: 26.2
elpy: 1.30.0

and my `.emacs` only this 4 lines:
```
(package-initialize)
(require 'ein)
(require 'ein-notebook)
(require 'ein-subpackages)
```
`ein:dev-pop-to-debug-iopub`got the result:
`Wrong type argument: ein:$notobook, nil`

and the report is :

## System info:

```cl
("EIN system info"
 :emacs-version "GNU Emacs 26.2 (build 1, arm-unknown-linux-androideabi)
 of 2019-06-09"
 :emacs-bzr-version nil
 :window-system nil
 :emacs-variant nil
 :os
 (:uname
  "Linux localhost 3.14.35 #1 SMP PREEMPT Tue Apr 10 18:39:29 JST 2018 armv7l Android
"
  :lsb-release nil)
 :notebook
 "Name: notebook
Version: 5.7.8
Summary: A web-based notebook environment for interactive computing
Home-page: http://jupyter.org
Author: Jupyter Development Team
Author-email: jupyter@googlegroups.com
License: BSD
Location: /data/data/com.termux/files/usr/lib/python3.7/site-packages
Requires: tornado, jupyter-core, jupyter-client, jinja2, nbconvert, traitlets, pyzmq, Send2Trash, ipython-genutils, terminado, prometheus-client, nbformat, ipykernel
Required-by: widgetsnbextension, jupyter
"
 :ipython "7.5.0
"
 :image-types nil
 :image-types-available nil
 :request
 (:backend url-retrieve)
 :ein
 (:version "20190605.2002"
	   :source-dir "/data/data/com.termux/files/home/.emacs.d/elpa/ein-20190605.2002/")
 :lib
 ((:name "websocket"
	 :path "~/.emacs.d/elpa/websocket-20190408.152/websocket.elc"
	 :featurep t
	 :version-var websocket-version
	 :version "1.10")
  (:name "auto-complete"
	 :path "~/.emacs.d/elpa/auto-complete-20170125.245/auto-complete.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "request"
	 :path "~/.emacs.d/elpa/request-20181129.1138/request.elc"
	 :featurep t
	 :version-var request-version
	 :version "0.3.0")
  (:name "deferred"
	 :path "~/.emacs.d/elpa/deferred-20170901.1330/deferred.elc"
	 :featurep t
	 :version-var deferred:version
	 :version "0.5.0")
  (:name "polymode"
	 :path "~/.emacs.d/elpa/polymode-20190605.2059/polymode.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "markdown-mode"
	 :path "~/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode.elc"
	 :featurep nil
	 :version-var nil
	 :version nil)
  (:name "dash"
	 :path "~/.emacs.d/elpa/dash-20190424.1804/dash.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "s"
	 :path "~/.emacs.d/elpa/s-20180406.808/s.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "skewer-mode"
	 :path "~/.emacs.d/elpa/skewer-mode-20180706.1807/skewer-mode.elc"
	 :featurep t
	 :version-var nil
	 :version nil)))
```

**Comment by yefeiyu** [comment]
Debugger entered--Lisp error: (void-function ein:crib-running-servers)
  (ein:crib-running-servers)
  eval((ein:crib-running-servers) nil)
  eval-expression((ein:crib-running-servers) nil nil 127)
  funcall-interactively(eval-expression (ein:crib-running-servers) nil nil 127)
  call-interactively(eval-expression nil nil)
  command-execute(eval-expression)

**Comment by dickmao** [comment]
You need `curl` in your path for a start.

Your `.emacs` should not need to explicitly require ein.  If you installed according to the README directions, then the autoload mechanism is supposed to handle all requires.

EIN was not intended to run on a mobile device.  I have no experience with termux, and never tested on it.

**Comment by yefeiyu** [comment]
"You need curl in your path for a start."
I could not understand.
Please give the step me, thank you

**Comment by dickmao** [comment]
pkg install curl

**Comment by yefeiyu** [comment]
I did install curl, but no use.
please read the 2nd reply, the result of `M-: (ein:crib-running-server)' and `jupyter notebook list --json`.
Is it wrong with other `packages`?

**Comment by dickmao** [comment]
I'm afraid I cannot help you with termux.

**Comment by yefeiyu** [comment]
Thank you for your answer and help.
Now I use `Linux Deploy`.
